src/test/mochitest/head.js

```diff
@@ -343,13 +344,13 @@ function isPaused(dbg) {
  * @static
  */
 async function waitForPaused(dbg) {
-  // We want to make sure that we get both a real paused event and
-  // that the state is fully populated. The client may do some more
-  // work (call other client methods) before populating the state.
-  let loading = waitForDispatch(dbg, "LOAD_OBJECT_PROPERTIES");
-  await waitForThreadEvents(dbg, "paused");
-  await waitForState(dbg, state => isPaused(dbg));
-  await loading;
+  const { getSelectedScope, hasLoadingObjects } = dbg.selectors;
+
+  return waitForState(
+    dbg,
+    state => isPaused(dbg) && getSelectedScope(state) && !hasLoadingObjects(state),
```

Comment:
we're now checking to see if we have a fetched scope and that there aren't any pending object fetches which is pretty nice